### ISSUE TITLE
Renaming the property of an inject service does not work

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/aspnetcore#40402]

**Comment by mkArtakMSFT** [comment]
@NTaylorMullen  can you please share more details about what specifically causes this and maybe you even know what needs to change to fix this? Thanks!

**Comment by NTaylorMullen** [comment]
@mkArtakMSFT absolutely. In this case the user has `@inject IEnumerable<int> enumerable`. At codegen time this generates two separate lambdas to power the IntelliSense:

```C#
__o = () => {
var
#line 1
enumerable
#line default
new object();
};
```

and

```C#
__o = () => {
varenumerable
#line default
__typehelper = default;
};
```

Note how neither one of these "helpers" actually reference the actual property which means if a user attempts to rename the parameter they wont actually have the intended effect
